src/text_formatter/text_formatter.ts

```diff
@@ -57,42 +47,39 @@ export function textToListFormat(input: String) {
  * @return a string-array where each element is one word
  */
 export function sentenceIntoWordList(input: String) {
-	const separator = /[ ,]+/; // List all the values where the content of a sentence will get splitted: whitespace[ ] and comma[,]
-	const splittedSentenceIntoListOfWords = input.split(separator);
-	const temp = [];
+
+	const splittedSentenceIntoListOfWords = input.split(wordSeparator);
+	const listOfWords = [];
 	for (let index = 0; index < splittedSentenceIntoListOfWords.length; index++) {
-		// remove all elements(senteces) which are a empty String after splitting
-		if (splittedSentenceIntoListOfWords[index].trim() == "") {
+		if (splittedSentenceIntoListOfWords[index].trim() == "") {	// remove all elements(senteces) which are a empty String after splitting
 			splittedSentenceIntoListOfWords.splice(index, 1);
 		} else {
-			temp.push(splittedSentenceIntoListOfWords[index].trim());
+			listOfWords.push(splittedSentenceIntoListOfWords[index].trim());
 		}
 	}
-	const ListOfWords = temp;
-	return ListOfWords;
+	return listOfWords;
 
 }
 
 /**
-* this function is the formatter for expressions; it calls "detectExpressionElements", "removeWhiteSpaces" and "replaceInputCaractersToReadablePrologCharacter";
+* this function is the formatter for expressions; it calls "formatExpressionElements", "removeWhiteSpaces" and "replaceInputCaractersToReadablePrologCharacter";
 * ---------------------------------------
 * @param a input string of an expression where all operator has been formatted
 * @return a string array where each element is a element of the expression
 */
 export function expressionFormatter(input: string) {
-	const detectedExpression = detectExpressionElements(input);
-	const separators = /(bracketLeft|bracketRight|equivalence|implication|negation|conjunction|disjunction)+/g;
-	const splittedExpression = detectedExpression.split(separators);
-	const temp = [];
+	const detectedExpression = formatExpressionElements(input);
+
+	const splittedExpression = detectedExpression.split(expressionSeparator);
```

Comment:
You wouldn't need to split here if you returned an array from `formatExpressionElements`.